main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,152 @@
+#include "main/lsp/requests/go_to_implementation.h"
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/json_types.h"
+#include "main/lsp/lsp.h"
+#include <cstddef>
+#include <memory>
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+namespace {
+struct MethodImplementationResults {
+    vector<core::Loc> locations;
+    unique_ptr<ResponseError> error;
+};
+} // namespace
+
+unique_ptr<ResponseError> makeInvalidParamsError(std::string error) {
+    return make_unique<ResponseError>((int)LSPErrorCodes::InvalidParams, error);
+}
+
+const MethodImplementationResults findMethodImplementations(const core::GlobalState &gs, core::SymbolRef method) {
+    if (!method.data(gs)->isMethod() || !method.data(gs)->isAbstract()) {
+        return {.error = {makeInvalidParamsError(
+                    "Go to implementation can be used only for methods or references of abstract classes")}};
+    }
```

Comment:
What about making a default-initialized `MethodImplementationResults` at the top of the method, and then mutating its fields before returning in each case?
```suggestion
    MethodImplementationResults res;
    if (!method.data(gs)->isMethod() || !method.data(gs)->isAbstract()) {
        res.error = makeInvalidParamsError("Go to implementation can be used only for methods or references of abstract classes");
        return res;
    }
```

This way you can avoid the initializer labels, and be more explicit about what's being returned here.

Reply:
Sounds great, looks much better this way